### PR TITLE
miniaudio: add version 0.11.11, support conan v2

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **miniaudio/***

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/README.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/miniaudio/all/conanfile.py [hunk 1]
```diff
@@ -25,66 +25,76 @@ class MiniaudioConan(ConanFile):
...
+        if self.options.header_only:
+            basic_layout(self, src_folder="src")
+        else:
+            cmake_layout(self, src_folder="src")
+
+    def package_id(self):
+        if self.options.header_only:
+            self.info.clear()
+            # TODO: for KB-H014. It must be removed when latest hooks/conan-center.py
+            self.info.header_only()
```
Comment (uilianries) [inline recipes/miniaudio/all/conanfile.py:53]:
```suggestion
```

fixed in Hooks! https://github.com/conan-io/hooks/issues/456

Comment (danimtb) [inline recipes/miniaudio/all/conanfile.py:53]:
```suggestion
```

**Review by danimtb — APPROVED** [review]

**Review by jwillikers — DISMISSED** [review]

File: recipes/miniaudio/all/conanfile.py [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
     description = "A single file audio playback and capture library."
-    topics = ("miniaudio", "header-only", "sound")
-    homepage = "https://github.com/mackron/miniaudio"
-    url = "https://github.com/conan-io/conan-center-index"
     license = ["Unlicense", "MIT-0"]
-    settings = "os", "compiler", "build_type", "arch"
-
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/mackron/miniaudio"
+    topics = ("miniaudio", "header-only", "sound")
```
Comment (prince-chrismc) [inline recipes/miniaudio/all/conanfile.py:15]: No need for the library name in the topic :)

```suggestion
    topics = ("audio" "header-only", "sound")
```

**Review by uilianries — DISMISSED** [review]

File: recipes/miniaudio/all/conanfile.py [hunk 3]
```diff
@@ -12,7 +12,7 @@ class MiniaudioConan(ConanFile):
     license = ["Unlicense", "MIT-0"]
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://github.com/mackron/miniaudio"
-    topics = ("miniaudio", "header-only", "sound")
+    topics = ("audio" "header-only", "sound")
```
Comment (jwillikers) [inline recipes/miniaudio/all/conanfile.py:15]: Just missing a comma here.

```suggestion
    topics = ("audio", "header-only", "sound")
```

**Review by jwillikers — CHANGES_REQUESTED** [review]

File: recipes/miniaudio/all/conanfile.py (hunk 3 above)
Comment (toge) [reply recipes/miniaudio/all/conanfile.py:15]: @jwillikers 
Thank you.
Just my mistake.

**Review by jwillikers — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**